File: removable/article/p_article_form.erl
-module(p_article_form).

-compile(export_all).

-include("sticky.hrl").
-include("article.hrl").

render(Method, Action, Article, Tags) ->
    Js = ale:ff("p_article_form.js"),
    ale:app_add_js(Js),

    Cancel = case Method of
        put -> [{a, [{href, ale:path(article, show, [Article#article.id])}], ?T("Cancel")}, " "];
        _   -> ""
    end,
    {form, [{id, article_form}, {method, post}, {action, Action}], [
        {input, [{type, hidden}, {name, "_method"}, {value, Method}]},

        {span, [{class, label}], ?T("Title")},
        {input, [{type, text}, {class, textbox}, {name, title}, {value, Article#article.title}]},

        {span, [{class, label}], ?T("Abstract")},
        {textarea, [{name, abstract}], Article#article.abstract},

        {span, [{class, label}], ?T("Body")},
        {textarea, [{name, body}], Article#article.body},

        h_tag:render_tag_selection(Tags),

        h_application:render_mathcha(),

        {input, [{type, submit}, {class, button}, {value, ?T("Save")}]}, " ", Cancel
    ]}.

File: removable/article/p_article_form.js
$('#article_form input[type=submit]').click(function() {
    $(this).hide();
    $(this).after('<img class="ajax-loader" src="/static/img/ajax-loader.gif" />');

    tinyMCE.triggerSave();

    var action   = $('#article_form').attr('action');
    var _method  = $('#article_form input[name=_method]').val();
    var title    = $('#article_form input[name=title]').val();
    var abstract = $('#article_form textarea[name=abstract]').val();
    var body     = $('#article_form textarea[name=body]').val();
    var tags     = $('#article_form input[name=tags]').val();
    var answer   = $('#article_form input[name=answer]').val();
    var eAnswer  = $('#article_form input[name=encrypted_answer]').val();
    var postData = {_method: _method, title: title, abstract: abstract, body: body, tags: tags, answer: answer, encrypted_answer: eAnswer};

    $.post(action, postData, function(data) {
        if (data.error) {
            alert(data.error);
            $('#article_form .ajax-loader').remove();
            $('#article_form input[type=submit]').show();
        } else {
            window.location.href = "/articles/" + data.atomic;
        }
    }, "json");
    return false;
});
